### PR TITLE
[master] properly return an error when exception raised in account creation

### PR DESCRIPTION
**Pull request by icehess**
When account creation failed, we don't return an error to UI, so UI doesn't know that account has not created, which result in UI continues to make other request to put limits, callflow an etc to the non-existent account.

### CODE REVIEW

File: applications/crossbar/src/modules/cb_accounts.erl [hunk 1]
```diff
@@ -1342,7 +1342,7 @@ set_notification_preference(Context, Preference) ->
 
 -spec create_account_mod(ne_binary()) -> any().
 create_account_mod(AccountId) ->
-    Db = kz_util:format_account_mod_id(AccountId),
+    Db = kz_util:format_account_mod_id(AccountId, raw),
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_accounts.erl]: this will fail, revert this line as the 2nd parameter is time related

Comment (icehess) [inline applications/crossbar/src/modules/cb_accounts.erl]: yeah, fixed, just did that for test :)

File: applications/crossbar/src/modules/cb_accounts.erl [hunk 2]
```diff
@@ -1342,8 +1342,7 @@ set_notification_preference(Context, Preference) ->
 
 -spec create_account_mod(ne_binary()) -> any().
 create_account_mod(AccountId) ->
-    Db = kz_util:format_account_mod_id(AccountId),
-    kazoo_modb:create(Db).
+    kazoo_modb:create(kazoo_modb:get_db(AccountId)).
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_accounts.erl]: why this change ? was it not working ?

Comment (icehess) [inline applications/crossbar/src/modules/cb_accounts.erl]: it's related to wht_util, `kazoo_modb:create` was expecting raw account_modb which would be passed to create_routines

Comment (icehess) [inline applications/crossbar/src/modules/cb_accounts.erl]: I know you fixed that in wht_util, but the main cause was here actually

Comment (lazedo) [inline applications/crossbar/src/modules/cb_accounts.erl]: and wasn't that fixed in wht_util ?

Comment (icehess) [inline applications/crossbar/src/modules/cb_accounts.erl]: yes, but with formating account_modb there is no need to handle encoded format in modb routines, since `cb_accounts` is the only place that calls `kazoo_modb:create`, and all the maybe_create functions in kazoo_modb converts the modb to raw. I would like to be consistence anywhere to not confused in the future.

Comment (lazedo) [inline applications/crossbar/src/modules/cb_accounts.erl]: wht_util had an an error that was fixed, the problem was not here, kz_services_modb handled the modb formats.

this pr is about returning a proper message in case account creation fails, lets keep the scope to that. 

to keep consistence, you can create another pr to remove usage of format_account_mod_id in all base code and deprecate it for 4.1 (add a note in announcements).

Comment (icehess) [inline applications/crossbar/src/modules/cb_accounts.erl]: ok, i'll open another pr for this later.
